From: Matt Benjamin <mbenjamin@redhat.com>
Date: Tue, 9 Aug 2016 16:49:41 -0400
Subject: rgw file: remove busy-wait in RGWLibFS::gc()

This is a background thread.  However, CPU is wasted.

Signed-off-by: Matt Benjamin <mbenjamin@redhat.com>
(cherry picked from commit ca33241286f52d849dbde8092507131b8b1108cc)

Fixes: http://tracker.ceph.com/issues/17321
---
 src/rgw/rgw_file.cc | 1 +
 src/rgw/rgw_file.h  | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/src/rgw/rgw_file.cc b/src/rgw/rgw_file.cc
index e454099..3ecd565 100644
--- a/src/rgw/rgw_file.cc
+++ b/src/rgw/rgw_file.cc
@@ -679,6 +679,7 @@ namespace rgw {
 	  } /* rgw_fh */
 	} /* event::type::READDIR */
       } /* ev */
+      std::this_thread::sleep_for(gc_interval);
     } while (! stop);
   } /* RGWLibFS::gc */
 
diff --git a/src/rgw/rgw_file.h b/src/rgw/rgw_file.h
index 7648b0b..deb9bd1 100644
--- a/src/rgw/rgw_file.h
+++ b/src/rgw/rgw_file.h
@@ -681,6 +681,9 @@ namespace rgw {
     struct rgw_fs fs;
     RGWFileHandle root_fh;
 
+    static constexpr std::chrono::seconds gc_interval =
+      std::chrono::seconds(120);
+
     mutable std::atomic<uint64_t> refcnt;
 
     RGWFileHandle::FHCache fh_cache;
